### ISSUE TITLE
[Bug-gmx_MMPBSA]: `print_res` in &decomp namelist not accept "all"

### ISSUE THREAD

**Issue by jsjyhzy** [Valdes-Tresanco-MS/gmx_MMPBSA#292]
### Bug summary

According to the document, the `print_res` in the `&decomp` namelist can be set as "all", however gmx_MMPBSA==1.5.6 is not accept “all“.

in GMXMMPBSA/utils.py, line 626, function `selector`, it seems the if statement only consider "within *" and residue selection, so is the option "all" is removed but not updated in document, or is it just a bug?

### Terminal output

```bash
[INFO   ] Starting gmx_MMPBSA v1.5.6
[INFO   ] Command-line
  mpirun -np 10 gmx_MMPBSA MPI -O -i mmpbsa.in -ci index.ndx -cg 0 1 -ct Docked_Protein_Ligand_Sample1-npt-prod.xtc -cs Docked_Protein_Ligand_Sample1-npt-prod.tpr -cp Docked_Protein_Ligand_Sample1.top -nogui -eo mmpbsa_ana.csv -deo mmpbsa_dec.csv

[INFO   ] Checking mmpbsa.in input file...
[INFO   ] Checking mmpbsa.in input file...Done.

[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/cpptraj
[INFO   ] tleap found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/tleap
[INFO   ] parmchk2 found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/parmchk2
[INFO   ] sander found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/sander
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER topologies from GROMACS files...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[INFO   ] Normal Complex: Saving group 0_1 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_COM.pdb
[INFO   ] No receptor structure file was defined. Using ST approach...
[INFO   ] Using receptor structure from complex to generate AMBER topology
[INFO   ] Normal Receptor: Saving group 0 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_REC.pdb
[INFO   ] No ligand structure file was defined. Using ST approach...
[INFO   ] Using ligand structure from complex to generate AMBER topology
[INFO   ] Normal Ligand: Saving group 1 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_LIG.pdb
[INFO   ] Checking the structures consistency...
[INFO   ] 
[INFO   ] Using topology conversion. Setting radiopt = 0...
[INFO   ] Building Normal Complex Amber topology...
[INFO   ] Detected Amber/OPLS force field topology format...
[WARNING] 6 invalid DIHEDRAL_PERIODICITY = 0 found in Complex topology... Setting DIHEDRAL_PERIODICITY = 1
[INFO   ] Assigning PBRadii mbondi2 to Complex...
[INFO   ] Writing Normal Complex AMBER topology...
[INFO   ] No Receptor topology file was defined. Using ST approach...
[INFO   ] Building AMBER Receptor topology from Complex...
[INFO   ] Assigning PBRadii mbondi2 to Receptor...
[INFO   ] Writing Normal Receptor AMBER topology...
[INFO   ] No Ligand topology file was defined. Using ST approach...
[INFO   ] Building AMBER Ligand topology from Complex...
[INFO   ] Assigning PBRadii mbondi2 to Ligand...
[INFO   ] Writing Normal Ligand AMBER topology...
[ERROR  ] MMPBSA_Error We expected something like this: A/2-10,35,41 B/104 but we get all instead.
           Check the gmx_MMPBSA.log file to report the problem.
  File "/public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/gmx_MMPBSA", line 8, in <module>
    sys.exit(gmxmmpbsa())
  File "/public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/lib/python3.9/site-packages/GMXMMPBSA/app.py", line 101, in gmxmmpbsa
    app.make_prmtops()
  File "/public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/lib/python3.9/site-packages/GMXMMPBSA/main.py", line 537, in make_prmtops
    self.FILES.mutant_receptor_prmtop, self.FILES.mutant_ligand_prmtop) = maketop.buildTopology()
  File "/public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 124, in buildTopology
    decomp_res = self.get_selected_residues(self.INPUT['print_res'])
  File "/public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 783, in get_selected_residues
    dist, res_selection = selector(select)
  File "/public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/lib/python3.9/site-packages/GMXMMPBSA/utils.py", line 626, in selector
    GMXMMPBSA_ERROR(f'We expected something like this: A/2-10,35,41 B/104 but we get {s} instead')
  File "/public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/lib/python3.9/site-packages/GMXMMPBSA/exceptions.py", line 169, in __init__
    raise exc(msg + '\nCheck the gmx_MMPBSA.log file to report the problem.')
MMPBSA_Error: We expected something like this: A/2-10,35,41 B/104 but we get all instead
Check the gmx_MMPBSA.log file to report the problem.
```

### gmx_MMPBSA.log

[INFO   ] Starting gmx_MMPBSA v1.5.6
[DEBUG  ] WDIR          : /public/home/neotrident/job/31/8a7212f2-4f4c-11ed-890e-27f472dd0509/rTze0WVj/Docked_Protein_Ligand_Sample1
[DEBUG  ] AMBERHOME     : /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1
[DEBUG  ] PYTHON EXE    : /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/python
[DEBUG  ] PYTHON VERSION: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:56:21) [GCC 10.3.0]
[DEBUG  ] MPI           : /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/mpirun
[DEBUG  ] ParmEd        : 3.4.3
[DEBUG  ] OS PLATFORM   : Linux-3.10.0-957.el7.x86_64-x86_64-with-glibc2.17
[DEBUG  ] OS SYSTEM     : Linux
[DEBUG  ] OS VERSION    : #1 SMP Fri Oct 25 22:34:08 UTC 2019
[DEBUG  ] OS PROCESSOR  : x86_64

[INFO   ] Command-line
  gmx_MMPBSA -O -i mmpbsa.in -ci index.ndx -cg 0 1 -ct Docked_Protein_Ligand_Sample1-npt-prod.xtc -cs Docked_Protein_Ligand_Sample1-npt-prod.tpr -cp Docked_Protein_Ligand_Sample1.top -nogui -eo mmpbsa_ana.csv -deo mmpbsa_dec.csv

[DEBUG  ] |Input file:
[DEBUG  ] |--------------------------------------------------------------
[DEBUG  ] |&general
[DEBUG  ] | startframe=1
[DEBUG  ] | endframe=999999
[DEBUG  ] | interval=1
[DEBUG  ] | temperature=298
[DEBUG  ] | exp_ki=0
[DEBUG  ] | PBRadii=3
[DEBUG  ] |
[DEBUG  ] |/
[DEBUG  ] |
[DEBUG  ] |&gb
[DEBUG  ] | igb=5
[DEBUG  ] | saltcon=0.0
[DEBUG  ] | surfoff=0.0
[DEBUG  ] | surften=0.0072
[DEBUG  ] |
[DEBUG  ] |/
[DEBUG  ] |
[DEBUG  ] |&decomp
[DEBUG  ] | dec_verbose=3
[DEBUG  ] | idecomp=4
[DEBUG  ] | print_res=all
[DEBUG  ] |
[DEBUG  ] |/
[DEBUG  ] |--------------------------------------------------------------
[DEBUG  ] 
[INFO   ] Checking mmpbsa.in input file...
[INFO   ] Checking mmpbsa.in input file...Done.

[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/cpptraj
[INFO   ] tleap found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/tleap
[INFO   ] parmchk2 found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/parmchk2
[INFO   ] sander found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin/sander
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER topologies from GROMACS files...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[DEBUG  ] Running command: echo name 0 GMXMMPBSA_REC\n name 1 GMXMMPBSA_LIG\n  0 | 1\n q\n | /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx make_ndx -n index.ndx -o _GMXMMPBSA_COM_index.ndx
[DEBUG  ]                 :-) GROMACS - gmx make_ndx, 2021.3-bioconda (-:
[DEBUG  ] 
[DEBUG  ]                             GROMACS is written by:
[DEBUG  ]      Andrey Alekseenko              Emile Apol              Rossen Apostolov     
[DEBUG  ]          Paul Bauer           Herman J.C. Berendsen           Par Bjelkmar       
[DEBUG  ]        Christian Blau           Viacheslav Bolnykh             Kevin Boyd        
[DEBUG  ]      Aldert van Buuren           Rudi van Drunen             Anton Feenstra      
[DEBUG  ]     Gilles Gouaillardet             Alan Gray               Gerrit Groenhof      
[DEBUG  ]        Anca Hamuraru            Vincent Hindriksen          M. Eric Irrgang      
[DEBUG  ]       Aleksei Iupinov           Christoph Junghans             Joe Jordan        
[DEBUG  ]     Dimitrios Karkoulis            Peter Kasson                Jiri Kraus        
[DEBUG  ]       Carsten Kutzner              Per Larsson              Justin A. Lemkul     
[DEBUG  ]        Viveca Lindahl            Magnus Lundborg             Erik Marklund       
[DEBUG  ]         Pascal Merz             Pieter Meulenhoff            Teemu Murtola       
[DEBUG  ]         Szilard Pall               Sander Pronk              Roland Schulz       
[DEBUG  ]        Michael Shirts            Alexey Shvetsov             Alfons Sijbers      
[DEBUG  ]        Peter Tieleman              Jon Vincent              Teemu Virolainen     
[DEBUG  ]      Christian Wennberg            Maarten Wolf              Artem Zhmurov       
[DEBUG  ]                            and the project leaders:
[DEBUG  ]         Mark Abraham, Berk Hess, Erik Lindahl, and David van der Spoel
[DEBUG  ] 
[DEBUG  ] Copyright (c) 1991-2000, University of Groningen, The Netherlands.
[DEBUG  ] Copyright (c) 2001-2019, The GROMACS development team at
[DEBUG  ] Uppsala University, Stockholm University and
[DEBUG  ] the Royal Institute of Technology, Sweden.
[DEBUG  ] check out http://www.gromacs.org for more information.
[DEBUG  ] 
[DEBUG  ] GROMACS is free software; you can redistribute it and/or modify it
[DEBUG  ] under the terms of the GNU Lesser General Public License
[DEBUG  ] as published by the Free Software Foundation; either version 2.1
[DEBUG  ] of the License, or (at your option) any later version.
[DEBUG  ] 
[DEBUG  ] GROMACS:      gmx make_ndx, version 2021.3-bioconda
[DEBUG  ] Executable:   /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx
[DEBUG  ] Data prefix:  /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1
[DEBUG  ] Working dir:  /public/home/neotrident/job/31/8a7212f2-4f4c-11ed-890e-27f472dd0509/rTze0WVj/Docked_Protein_Ligand_Sample1
[DEBUG  ] Command line:
[DEBUG  ]   gmx make_ndx -n index.ndx -o _GMXMMPBSA_COM_index.ndx
[DEBUG  ] 
[DEBUG  ] 
[DEBUG  ] GROMACS reminds you: "Naive you are if you believe life favours those who aren't naive." (Piet Hein)
[DEBUG  ] 
[DEBUG  ] Going to read 1 old index file(s)
[DEBUG  ] Counted atom numbers up to 4382 in index file
[DEBUG  ] 
[DEBUG  ]   0 new_group_0         :  4346 atoms
[DEBUG  ]   1 new_group_1         :    36 atoms
[DEBUG  ] 
[DEBUG  ]  nr : group      '!': not  'name' nr name   'splitch' nr    Enter: list groups
[DEBUG  ]  'a': atom       '&': and  'del' nr         'splitres' nr   'l': list residues
[DEBUG  ]  't': atom type  '|': or   'keep' nr        'splitat' nr    'h': help
[DEBUG  ]  'r': residue              'res' nr         'chain' char
[DEBUG  ]  "name": group             'case': case sensitive           'q': save and quit
[DEBUG  ]  'ri': residue index
[DEBUG  ] 
[DEBUG  ] > 
[DEBUG  ] 
[DEBUG  ] > 
[DEBUG  ] 
[DEBUG  ] > 
[DEBUG  ] Copied index group 0 'GMXMMPBSA_REC'
[DEBUG  ] Copied index group 1 'GMXMMPBSA_LIG'
[DEBUG  ] Merged two groups with OR: 4346 36 -> 4382
[DEBUG  ] 
[DEBUG  ]   2 GMXMMPBSA_REC_GMXMMPBSA_LIG:  4382 atoms
[DEBUG  ] 
[DEBUG  ] > 
[INFO   ] Normal Complex: Saving group 0_1 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_COM.pdb
[DEBUG  ] Running command: echo GMXMMPBSA_REC_GMXMMPBSA_LIG | /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx trjconv -f Docked_Protein_Ligand_Sample1-npt-prod.xtc -s Docked_Protein_Ligand_Sample1-npt-prod.tpr -o _GMXMMPBSA_COM.pdb -n _GMXMMPBSA_COM_index.ndx -dump 0
[DEBUG  ]                  :-) GROMACS - gmx trjconv, 2021.3-bioconda (-:
[DEBUG  ] 
[DEBUG  ]                             GROMACS is written by:
[DEBUG  ]      Andrey Alekseenko              Emile Apol              Rossen Apostolov     
[DEBUG  ]          Paul Bauer           Herman J.C. Berendsen           Par Bjelkmar       
[DEBUG  ]        Christian Blau           Viacheslav Bolnykh             Kevin Boyd        
[DEBUG  ]      Aldert van Buuren           Rudi van Drunen             Anton Feenstra      
[DEBUG  ]     Gilles Gouaillardet             Alan Gray               Gerrit Groenhof      
[DEBUG  ]        Anca Hamuraru            Vincent Hindriksen          M. Eric Irrgang      
[DEBUG  ]       Aleksei Iupinov           Christoph Junghans             Joe Jordan        
[DEBUG  ]     Dimitrios Karkoulis            Peter Kasson                Jiri Kraus        
[DEBUG  ]       Carsten Kutzner              Per Larsson              Justin A. Lemkul     
[DEBUG  ]        Viveca Lindahl            Magnus Lundborg             Erik Marklund       
[DEBUG  ]         Pascal Merz             Pieter Meulenhoff            Teemu Murtola       
[DEBUG  ]         Szilard Pall               Sander Pronk              Roland Schulz       
[DEBUG  ]        Michael Shirts            Alexey Shvetsov             Alfons Sijbers      
[DEBUG  ]        Peter Tieleman              Jon Vincent              Teemu Virolainen     
[DEBUG  ]      Christian Wennberg            Maarten Wolf              Artem Zhmurov       
[DEBUG  ]                            and the project leaders:
[DEBUG  ]         Mark Abraham, Berk Hess, Erik Lindahl, and David van der Spoel
[DEBUG  ] 
[DEBUG  ] Copyright (c) 1991-2000, University of Groningen, The Netherlands.
[DEBUG  ] Copyright (c) 2001-2019, The GROMACS development team at
[DEBUG  ] Uppsala University, Stockholm University and
[DEBUG  ] the Royal Institute of Technology, Sweden.
[DEBUG  ] check out http://www.gromacs.org for more information.
[DEBUG  ] 
[DEBUG  ] GROMACS is free software; you can redistribute it and/or modify it
[DEBUG  ] under the terms of the GNU Lesser General Public License
[DEBUG  ] as published by the Free Software Foundation; either version 2.1
[DEBUG  ] of the License, or (at your option) any later version.
[DEBUG  ] 
[DEBUG  ] GROMACS:      gmx trjconv, version 2021.3-bioconda
[DEBUG  ] Executable:   /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx
[DEBUG  ] Data prefix:  /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1
[DEBUG  ] Working dir:  /public/home/neotrident/job/31/8a7212f2-4f4c-11ed-890e-27f472dd0509/rTze0WVj/Docked_Protein_Ligand_Sample1
[DEBUG  ] Command line:
[DEBUG  ]   gmx trjconv -f Docked_Protein_Ligand_Sample1-npt-prod.xtc -s Docked_Protein_Ligand_Sample1-npt-prod.tpr -o _GMXMMPBSA_COM.pdb -n _GMXMMPBSA_COM_index.ndx -dump 0
[DEBUG  ] 
[DEBUG  ] Will write pdb: Protein data bank file
[DEBUG  ] Reading file Docked_Protein_Ligand_Sample1-npt-prod.tpr, VERSION 2022 (single precision)
[DEBUG  ] Reading file Docked_Protein_Ligand_Sample1-npt-prod.tpr, VERSION 2022 (single precision)
[DEBUG  ] Group     0 (  GMXMMPBSA_REC) has  4346 elements
[DEBUG  ] Group     1 (  GMXMMPBSA_LIG) has    36 elements
[DEBUG  ] Group     2 (GMXMMPBSA_REC_GMXMMPBSA_LIG) has  4382 elements
[DEBUG  ] Select a group: 
Reading frame       0 time    0.000   
[DEBUG  ] Precision of Docked_Protein_Ligand_Sample1-npt-prod.xtc is 0.001 (nm)
[DEBUG  ] 
Reading frame       1 time    0.020   
Reading frame       0 time    0.000   
[DEBUG  ] Dumping frame at t= 0 ps
[DEBUG  ] 
[DEBUG  ] GROMACS reminds you: "Don't You Wish You Never Met Her, Dirty Blue Gene?" (Captain Beefheart)
[DEBUG  ] 
[DEBUG  ] Note that major changes are planned in future for trjconv, to improve usability and utility.
[DEBUG  ] Select group for output
[DEBUG  ] Selected 2: 'GMXMMPBSA_REC_GMXMMPBSA_LIG'
[INFO   ] No receptor structure file was defined. Using ST approach...
[INFO   ] Using receptor structure from complex to generate AMBER topology
[INFO   ] Normal Receptor: Saving group 0 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_REC.pdb
[DEBUG  ] Running command: echo 0 | /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx trjconv -f Docked_Protein_Ligand_Sample1-npt-prod.xtc -s Docked_Protein_Ligand_Sample1-npt-prod.tpr -o _GMXMMPBSA_REC.pdb -n _GMXMMPBSA_COM_index.ndx -dump 0
[DEBUG  ]                  :-) GROMACS - gmx trjconv, 2021.3-bioconda (-:
[DEBUG  ] 
[DEBUG  ]                             GROMACS is written by:
[DEBUG  ]      Andrey Alekseenko              Emile Apol              Rossen Apostolov     
[DEBUG  ]          Paul Bauer           Herman J.C. Berendsen           Par Bjelkmar       
[DEBUG  ]        Christian Blau           Viacheslav Bolnykh             Kevin Boyd        
[DEBUG  ]      Aldert van Buuren           Rudi van Drunen             Anton Feenstra      
[DEBUG  ]     Gilles Gouaillardet             Alan Gray               Gerrit Groenhof      
[DEBUG  ]        Anca Hamuraru            Vincent Hindriksen          M. Eric Irrgang      
[DEBUG  ]       Aleksei Iupinov           Christoph Junghans             Joe Jordan        
[DEBUG  ]     Dimitrios Karkoulis            Peter Kasson                Jiri Kraus        
[DEBUG  ]       Carsten Kutzner              Per Larsson              Justin A. Lemkul     
[DEBUG  ]        Viveca Lindahl            Magnus Lundborg             Erik Marklund       
[DEBUG  ]         Pascal Merz             Pieter Meulenhoff            Teemu Murtola       
[DEBUG  ]         Szilard Pall               Sander Pronk              Roland Schulz       
[DEBUG  ]        Michael Shirts            Alexey Shvetsov             Alfons Sijbers      
[DEBUG  ]        Peter Tieleman              Jon Vincent              Teemu Virolainen     
[DEBUG  ]      Christian Wennberg            Maarten Wolf              Artem Zhmurov       
[DEBUG  ]                            and the project leaders:
[DEBUG  ]         Mark Abraham, Berk Hess, Erik Lindahl, and David van der Spoel
[DEBUG  ] 
[DEBUG  ] Copyright (c) 1991-2000, University of Groningen, The Netherlands.
[DEBUG  ] Copyright (c) 2001-2019, The GROMACS development team at
[DEBUG  ] Uppsala University, Stockholm University and
[DEBUG  ] the Royal Institute of Technology, Sweden.
[DEBUG  ] check out http://www.gromacs.org for more information.
[DEBUG  ] 
[DEBUG  ] GROMACS is free software; you can redistribute it and/or modify it
[DEBUG  ] under the terms of the GNU Lesser General Public License
[DEBUG  ] as published by the Free Software Foundation; either version 2.1
[DEBUG  ] of the License, or (at your option) any later version.
[DEBUG  ] 
[DEBUG  ] GROMACS:      gmx trjconv, version 2021.3-bioconda
[DEBUG  ] Executable:   /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx
[DEBUG  ] Data prefix:  /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1
[DEBUG  ] Working dir:  /public/home/neotrident/job/31/8a7212f2-4f4c-11ed-890e-27f472dd0509/rTze0WVj/Docked_Protein_Ligand_Sample1
[DEBUG  ] Command line:
[DEBUG  ]   gmx trjconv -f Docked_Protein_Ligand_Sample1-npt-prod.xtc -s Docked_Protein_Ligand_Sample1-npt-prod.tpr -o _GMXMMPBSA_REC.pdb -n _GMXMMPBSA_COM_index.ndx -dump 0
[DEBUG  ] 
[DEBUG  ] Will write pdb: Protein data bank file
[DEBUG  ] Reading file Docked_Protein_Ligand_Sample1-npt-prod.tpr, VERSION 2022 (single precision)
[DEBUG  ] Reading file Docked_Protein_Ligand_Sample1-npt-prod.tpr, VERSION 2022 (single precision)
[DEBUG  ] Group     0 (  GMXMMPBSA_REC) has  4346 elements
[DEBUG  ] Group     1 (  GMXMMPBSA_LIG) has    36 elements
[DEBUG  ] Group     2 (GMXMMPBSA_REC_GMXMMPBSA_LIG) has  4382 elements
[DEBUG  ] Select a group: 
Reading frame       0 time    0.000   
[DEBUG  ] Precision of Docked_Protein_Ligand_Sample1-npt-prod.xtc is 0.001 (nm)
[DEBUG  ] 
Reading frame       1 time    0.020   
Reading frame       0 time    0.000   
[DEBUG  ] Dumping frame at t= 0 ps
[DEBUG  ] 
[DEBUG  ] GROMACS reminds you: "Don't You Wish You Never Met Her, Dirty Blue Gene?" (Captain Beefheart)
[DEBUG  ] 
[DEBUG  ] Note that major changes are planned in future for trjconv, to improve usability and utility.
[DEBUG  ] Select group for output
[DEBUG  ] Selected 0: 'GMXMMPBSA_REC'
[INFO   ] No ligand structure file was defined. Using ST approach...
[INFO   ] Using ligand structure from complex to generate AMBER topology
[INFO   ] Normal Ligand: Saving group 1 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_LIG.pdb
[DEBUG  ] Running command: echo 1 | /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx trjconv -f Docked_Protein_Ligand_Sample1-npt-prod.xtc -s Docked_Protein_Ligand_Sample1-npt-prod.tpr -o _GMXMMPBSA_LIG.pdb -n _GMXMMPBSA_COM_index.ndx -dump 0
[DEBUG  ]                  :-) GROMACS - gmx trjconv, 2021.3-bioconda (-:
[DEBUG  ] 
[DEBUG  ]                             GROMACS is written by:
[DEBUG  ]      Andrey Alekseenko              Emile Apol              Rossen Apostolov     
[DEBUG  ]          Paul Bauer           Herman J.C. Berendsen           Par Bjelkmar       
[DEBUG  ]        Christian Blau           Viacheslav Bolnykh             Kevin Boyd        
[DEBUG  ]      Aldert van Buuren           Rudi van Drunen             Anton Feenstra      
[DEBUG  ]     Gilles Gouaillardet             Alan Gray               Gerrit Groenhof      
[DEBUG  ]        Anca Hamuraru            Vincent Hindriksen          M. Eric Irrgang      
[DEBUG  ]       Aleksei Iupinov           Christoph Junghans             Joe Jordan        
[DEBUG  ]     Dimitrios Karkoulis            Peter Kasson                Jiri Kraus        
[DEBUG  ]       Carsten Kutzner              Per Larsson              Justin A. Lemkul     
[DEBUG  ]        Viveca Lindahl            Magnus Lundborg             Erik Marklund       
[DEBUG  ]         Pascal Merz             Pieter Meulenhoff            Teemu Murtola       
[DEBUG  ]         Szilard Pall               Sander Pronk              Roland Schulz       
[DEBUG  ]        Michael Shirts            Alexey Shvetsov             Alfons Sijbers      
[DEBUG  ]        Peter Tieleman              Jon Vincent              Teemu Virolainen     
[DEBUG  ]      Christian Wennberg            Maarten Wolf              Artem Zhmurov       
[DEBUG  ]                            and the project leaders:
[DEBUG  ]         Mark Abraham, Berk Hess, Erik Lindahl, and David van der Spoel
[DEBUG  ] 
[DEBUG  ] Copyright (c) 1991-2000, University of Groningen, The Netherlands.
[DEBUG  ] Copyright (c) 2001-2019, The GROMACS development team at
[DEBUG  ] Uppsala University, Stockholm University and
[DEBUG  ] the Royal Institute of Technology, Sweden.
[DEBUG  ] check out http://www.gromacs.org for more information.
[DEBUG  ] 
[DEBUG  ] GROMACS is free software; you can redistribute it and/or modify it
[DEBUG  ] under the terms of the GNU Lesser General Public License
[DEBUG  ] as published by the Free Software Foundation; either version 2.1
[DEBUG  ] of the License, or (at your option) any later version.
[DEBUG  ] 
[DEBUG  ] GROMACS:      gmx trjconv, version 2021.3-bioconda
[DEBUG  ] Executable:   /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1/bin.AVX2_256/gmx
[DEBUG  ] Data prefix:  /public/home/neotrident/miniconda3/envs/gmxMMPBSA-1.5.6-rc1
[DEBUG  ] Working dir:  /public/home/neotrident/job/31/8a7212f2-4f4c-11ed-890e-27f472dd0509/rTze0WVj/Docked_Protein_Ligand_Sample1
[DEBUG  ] Command line:
[DEBUG  ]   gmx trjconv -f Docked_Protein_Ligand_Sample1-npt-prod.xtc -s Docked_Protein_Ligand_Sample1-npt-prod.tpr -o _GMXMMPBSA_LIG.pdb -n _GMXMMPBSA_COM_index.ndx -dump 0
[DEBUG  ] 
[DEBUG  ] Will write pdb: Protein data bank file
[DEBUG  ] Reading file Docked_Protein_Ligand_Sample1-npt-prod.tpr, VERSION 2022 (single precision)
[DEBUG  ] Reading file Docked_Protein_Ligand_Sample1-npt-prod.tpr, VERSION 2022 (single precision)
[DEBUG  ] Group     0 (  GMXMMPBSA_REC) has  4346 elements
[DEBUG  ] Group     1 (  GMXMMPBSA_LIG) has    36 elements
[DEBUG  ] Group     2 (GMXMMPBSA_REC_GMXMMPBSA_LIG) has  4382 elements
[DEBUG  ] Select a group: 
Reading frame       0 time    0.000   
[DEBUG  ] Precision of Docked_Protein_Ligand_Sample1-npt-prod.xtc is 0.001 (nm)
[DEBUG  ] 
Reading frame       1 time    0.020   
Reading frame       0 time    0.000   
[DEBUG  ] Dumping frame at t= 0 ps
[DEBUG  ] 
[DEBUG  ] GROMACS reminds you: "Don't You Wish You Never Met Her, Dirty Blue Gene?" (Captain Beefheart)
[DEBUG  ] 
[DEBUG  ] Note that major changes are planned in future for trjconv, to improve usability and utility.
[DEBUG  ] Select group for output
[DEBUG  ] Selected 1: 'GMXMMPBSA_LIG'
[INFO   ] Checking the structures consistency...
[INFO   ] 
[INFO   ] Using topology conversion. Setting radiopt = 0...
[INFO   ] Building Normal Complex Amber topology...
[INFO   ] Detected Amber/OPLS force field topology format...
[WARNING] 6 invalid DIHEDRAL_PERIODICITY = 0 found in Complex topology... Setting DIHEDRAL_PERIODICITY = 1
[INFO   ] Assigning PBRadii mbondi2 to Complex...
[INFO   ] Writing Normal Complex AMBER topology...
[INFO   ] No Receptor topology file was defined. Using ST approach...
[INFO   ] Building AMBER Receptor topology from Complex...
[INFO   ] Assigning PBRadii mbondi2 to Receptor...
[INFO   ] Writing Normal Receptor AMBER topology...
[INFO   ] No Ligand topology file was defined. Using ST approach...
[INFO   ] Building AMBER Ligand topology from Complex...
[INFO   ] Assigning PBRadii mbondi2 to Ligand...
[INFO   ] Writing Normal Ligand AMBER topology...
[ERROR  ] MMPBSA_Error We expected something like this: A/2-10,35,41 B/104 but we get all instead.
           Check the gmx_MMPBSA.log file to report the problem.

### gmx_MMPBSA Version

gmx_MMPBSA v1.5.6 based on MMPBSA version 16.0 and AmberTools 20

### Python version

Python 3.9.13

### Installation

conda AmberTools + pip

**Comment by Valdes-Tresanco-MS** [comment]
It is definitely a bug. I will fix it ASAP

**Comment by Valdes-Tresanco-MS** [comment]
Please update as follow:
```bash
python -m pip install git+https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA.git - U
```
Let me know if works for you

**Comment by jsjyhzy** [comment]
Many thanks, it works now

**Comment by jsjyhzy** [comment]
Many thanks, it works now

**Comment by allanlinzhang** [comment]
> Please update as follow:
> 
> ```shell
> python -m pip install git+https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA.git - U
> ```
> 
> Let me know if works for you

I got the error  "[ERROR  ] MMPBSA_Error 

We expected something like this: A/2-10,35,41 B/104 but we get wihin instead" when I ran the command  "gmx_MMPBSA -O -i gmx_MMPBSA.in -cs md_0_1.tpr -cp topol.top -ci index.ndx -cg 18 19 -ct trj_25-30ns.xtc -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv".
But I can't figure it out when follow your guide.
Some way else to solve the problem?
